### PR TITLE
docs: add make-disk-image

### PR DESCRIPTION
**Pull request by Janik-Haag**
Just adding some docs for `lib/make-disk-image`

### CODE REVIEW

File: docs/reference.md [hunk 1]
```diff
@@ -44,4 +44,46 @@ Options:
...
+* --post-format-files <src> <dst>
+  copies the src to the dst on the finished image
+  These end up in the images later and is useful if you want to add some extra stateful files
+  They will have the same permissions but will be owned by root:root
+* --build-memory
+  specify the ammount of memory that gets allocated to the build vm (in mb)
+  This can be usefull if you want to build images with a more involed NixOS config
+  By default the vm will get 1024M/1GB
+* --write-to-disk </dev/disk>
+  use a actuall disk instead of writing to a file
```
Comment (Lassulus) [inline docs/reference.md:73]:
```suggestion
  use an actual disk instead of writing to a file
```

File: docs/reference.md [hunk 2]
```diff
@@ -44,4 +44,70 @@ Options:
...
+# Generating Disk Images with Secrets Included in Disko
+
+Using Disko on NixOS allows you to efficiently create `.raw` VM images from a
+system configuration. The generated image can be used as a VM or directly
+written to a physical drive to create a bootable disk. Follow the steps below to
+generate disk images:
+
+## Generating the `.raw` VM Image
+
+1. **Modify the script name:** Replace `mySystem` in the command below with your
```
Comment (Janik-Haag) [inline docs/reference.md:56]: I feel like script name doesn't really fit here but don't know a better phrasing

File: docs/reference.md [hunk 3]
```diff
@@ -44,4 +44,70 @@ Options:
...
+     By default the vm will get 1024M/1GB
+   * --write-to-disk </dev/disk>
+     use an actuall disk instead of writing to a file
+     This only works if your conifg has only one disk specified
+     There is no check if the specified path is actually a disk so you can also write to another file
+   ```
+
+## Additional Configuration
+
+- For virtual drive use, define the image size in your Disko configuration:
```
Comment (Janik-Haag) [inline docs/reference.md:88]:
```suggestion
- For virtual drives, define the image size in your Disko configuration:
```

File: docs/reference.md [hunk 4]
```diff
@@ -44,4 +44,70 @@ Options:
...
+
+  ```bash
+  disko.devices.disk.<drive>.imageSize = "32G"; # Set your preferred size
+  ```
+
+- If the `.raw` image size is problematic, use `--write-to-disk` to write
+  directly to a drive. This bypasses the `.raw` file generation and is suitable
+  for single disk, simple setups.
```
Comment (Janik-Haag) [inline docs/reference.md:96]:
```suggestion
- If the `.raw` image size is not optimal, use `--write-to-disk` to write
  directly to a drive. This bypasses the `.raw` file generation, which saves on read/write operations
  and is suitable for single disk setups.
```

File: docs/reference.md (hunk 3 above)
Comment (Mic92) [reply docs/reference.md:88]: technically the other one is also correct, but maybe more confusing.

File: docs/reference.md [hunk 5]
```diff
@@ -44,4 +44,70 @@ Options:
   run with set -x
 ```
 
-## 
+# Generating Disk Images with Secrets Included in Disko
```
[the remark this answers is not in the document]
Comment (Mic92) [reply docs/reference.md:47]: I think both ways are fine here as well, but we can stick with your variant.

File: docs/reference.md (hunk 2 above)
Comment (Mic92) [reply docs/reference.md:56]:
```suggestion
1. **Build the disko image script:** Replace `mySystem` in the command below with your
```

**Comment by Mic92** [comment]
@mergify queue

**Comment by Lassulus** [comment]
@mergify queue

**Pull request merged**